src/nxt_http_variables.c

```diff
@@ -77,3 +83,13 @@ nxt_http_var_host(nxt_task_t *task, nxt_var_query_t *query, nxt_str_t *str,
 
     return NXT_OK;
 }
+
+
+static nxt_int_t
+nxt_http_var_dollar(nxt_task_t *task, nxt_var_query_t *query, nxt_str_t *str,
+    void *ctx)
+{
+    *str = (nxt_str_t) nxt_string("$");
```

Comment:
Maybe we can use here `nxt_str_set()` to avoid the cast?

```c
alx@asus5775:~/src/nginx/unit$ grepc nxt_string


./src/nxt_string.h:108:

#define nxt_string(str)       { nxt_length(str), (u_char *) str }
alx@asus5775:~/src/nginx/unit$ grepc nxt_str_set


./src/nxt_string.h:113:

#define nxt_str_set(str, text)                                                \
    do {                                                                      \
        (str)->length = nxt_length(text);                                     \
        (str)->start = (u_char *) text;                                       \
    } while (0)
```

Reply:
Good catch!